R/sample_swap.R

```diff
@@ -126,8 +126,14 @@ sampleSwaps = function(bams = NULL, build = "hg19", prefix = NULL, add = TRUE, m
   sample_matches = sample_matches[1:(length(sample_matches)-1)]
 
   pos_mathces = lapply(sample_matches, function(sample_pair){
-    if(nrow(sample_pair) > 0){
-      unique(unlist(sample_pair[XY_possibly_paired == "Yes", .(X_bam, Y_bam)], use.names = FALSE))
+    if (all(c("XY_possibly_paired", "X_bam", "Y_bam") %in% colnames(sample_pair))) {
+      # Filter out rows where X_bam or Y_bam have length zero
+      valid_rows <- sapply(sample_pair$X_bam, nchar) > 0 & sapply(sample_pair$Y_bam, nchar) > 0
+      filtered_sample_pair <- sample_pair[valid_rows, ]
+      
+      if(nrow(filtered_sample_pair) > 0){
+        unique(unlist(filtered_sample_pair[XY_possibly_paired == "Yes", .(X_bam, Y_bam)], use.names = FALSE))
+      }
     }
   })
   pos_mathces = pos_mathces[which(lapply(pos_mathces, function(x) length(x) > 0) == TRUE)]
```

Comment:
It should be better to use `pos_mathces[sapply(pos_mathces, function(x) length(x) > 0)]`?

Reply:
Yes, you are right. That works as well